From d950e2daa770c22b5cfddd00f4323574ab0607fb Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@nxp.com>
Date: Thu, 27 Jul 2017 00:19:55 +0800
Subject: [PATCH 2291/5242] MLK-16013-12: staging: typec: tcpci: clear pending
 events before request irq

commit  0b5a180023d933ff6f07a92896d02130c6f18e43 from
https://source.codeaurora.org/external/imx/linux-imx.git

Clear all possible pending events before request irq.

Reviewed-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Li Jun <jun.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/staging/typec/tcpci.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/staging/typec/tcpci.c b/drivers/staging/typec/tcpci.c
index 76492c2..eeaa8ab 100644
--- a/drivers/staging/typec/tcpci.c
+++ b/drivers/staging/typec/tcpci.c
@@ -561,6 +561,8 @@ static int tcpci_probe(struct i2c_client *client,
 	if (IS_ERR(chip->data.regmap))
 		return PTR_ERR(chip->data.regmap);
 
+	/* Clear and disable chip interrupts */
+	tcpci_write16(tcpci, TCPC_ALERT, 0xffff);
 	/* Disable chip interrupts before requesting irq */
 	err = regmap_raw_write(chip->data.regmap, TCPC_ALERT_MASK, &val,
 			       sizeof(u16));
-- 
1.7.9.5

